src/alloc/AllocBenchmarks.jl

```diff
@@ -0,0 +1,58 @@
+module AllocBenchmarks
+
+using BenchmarkTools
+
+const SUITE = BenchmarkGroup()
+
+function perf_alloc_many_arrays()
+    for _ in 1:10000000
+        # global to ensure that this is heap allocated
+        global a = []
```

Comment:
Maybe easier to escape into a:

```
@noinline escape(x) = (Base.inferencebarrier(x); return nothing)
```

Reply:
Huh, in place of `global a = []`? Not quite sure what the `escape` and `inferencebarrier` are doing here

Reply:
That is just a pure function that returns `nothing`–we don't need to infer that to remove it

Reply:
I'm also confused by this. Are you saying this is the suggested diff?:
```suggestion
        escape([])
```
with your function defined elsewhere?

Does that inferencebarrier actually prevent the compiler from compiling away the function call, even though `escape()` ultimately returns `nothing`?

-----------

Also why is this preferable to what @vilterp has already? (🙏 Plz teach me to write good benchmarks 🙏)